scripts/build-ancillary-repository.sh

```diff
@@ -57,7 +57,8 @@ function resolve_s3_uri() {
 		resolved_uri="s3://$bucket/$pkg_prefix"
 	elif [[ -n "$latest_subprefix" ]]; then
 		aws s3 cp --quiet \
-			"s3://$bucket/builds/$jenkinsid/$latest_subprefix" .
+			"s3://$bucket/builds/$jenkinsid/$latest_subprefix" . ||
+			exit 1
```

Comment:
I had a failure there which wasn't caught by `errexit`. As others have pointed out `errexit` doesn't seem to be reliable. As an example, see http://collins.d.delphix.com:21968/job/devops-gate/job/master/job/appliance-build/job/master/job/pre-push/3/console, grep for `cat: latest: No such file or directory`.

Reply:
I did some simple testing, and yes, it looks like `set -o` is not preserved across subshells (e.g. when using `$(...)`).

I think we should remove this change from the PR since it's not really related; we have this issue across the entire script, so we should fix it for all cases if we're going to fix it for any one case.

Reply:
Okay, can you open an issue to track this?

Reply:
Yup, I'll do that now.